### ISSUE TITLE
Feature Request/Idea: Make sure that for datasets published with CC0 waiver or standard license, metadata exports include the waiver or license

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse#8798]
**Overview of the Feature Request**
Make sure that for datasets published with a CC0 waiver or a standard license, their metadata exports include the waiver or license.

**What kind of user is the feature intended for?**
Curator, Guest

**What inspired the request?**
While reviewing the metadata exports of datasets published in repositories running a Dataverse software version that includes the "multiple license" update, I noticed that for certain datasets published before that update, the metadata exports don't include the CC0 waiver or other standard license.

For example, after the multiple license update, datasets published before that update with a CC0 waiver and metadata entered in one or more "Terms of Use" fields were updated to include text in the Terms of Use field that read that "This dataset is made available under a Creative Commons CC0 license with the following additional/modified terms and conditions": 

<kbd><img width="1163" alt="Screen Shot 2022-06-13 at 12 34 28 PM" src="https://user-images.githubusercontent.com/18374574/173401837-cb460302-8d3f-4dd2-a979-ddb8d824a241.png"></kbd>

But the OpenAIRE and Schema.org metadata exports of those datasets do not include information about the CC0 waiver:
- [OpenAIRE export](https://dataverse.harvard.edu/api/datasets/export?exporter=oai_datacite&persistentId=doi:10.7910/DVN/WHNXKY) of dataset with CC0 waiver plus metadata entered in one or more "Terms of Use" fields
- [Schema.org export](https://dataverse.harvard.edu/api/datasets/export?exporter=schema.org&persistentId=doi:10.7910/DVN/WHNXKY) of dataset with CC0 waiver plus metadata entered in one or more "Terms of Use" fields

For comparison, see the OpenAIRE and Schema.org metadata exports of a dataset with a CC0 waiver and nothing entered in any of the "Terms of Use" fields:

- [OpenAIRE export](https://dataverse.harvard.edu/api/datasets/export?exporter=oai_datacite&persistentId=doi%3A10.7910/DVN/SN869P) of dataset with CC0 waiver and no metadata entered in one or more "Terms of Use" fields
  <kbd><img width="737" alt="Screen Shot 2022-06-13 at 12 43 14 PM" src="https://user-images.githubusercontent.com/18374574/173403522-0772685f-417b-4f11-9cfb-24cddcf7f8f7.png"></kbd>

- [Schema.org export](https://dataverse.harvard.edu/api/datasets/export?exporter=schema.org&persistentId=doi%3A10.7910/DVN/SN869P) of dataset with CC0 waiver and no metadata entered in one or more "Terms of Use" fields
  <kbd><img width="485" alt="Screen Shot 2022-06-13 at 12 42 50 PM" src="https://user-images.githubusercontent.com/18374574/173403637-850c5f74-06b4-4f83-92ac-7d3308284716.png"></kbd>

**What existing behavior do you want changed?**
For datasets that were published with a CC0 waiver (or possibly a CC-BY license for some Dataverse repositories whose default license was CC-BY), include in their metadata exports information about those waivers or licenses. This will become more important as license metadata is included in more exports (e.g. being sent to DataCite (https://github.com/IQSS/dataverse/issues/5889)) and is indexed and made searchable (e.g. https://github.com/IQSS/dataverse/issues/7482).

**Comment by qqmyers** [comment]
FWIW: The design discussion around this was specifically to not reference the standard license (or consider the dataset to be using that license) when there are custom terms (any field that appears when you pick custom terms). 

I think the schema.org example is what was intended i.e. there is still a `"license": "https://dataverse.harvard.edu/api/datasets/:persistentId/versions/2.1/customlicense?persistentId=doi:10.7910/DVN/WHNXKY"`, link in the metadata. That should be similar for other formats where the URI was sent before - we now send the custom terms/licenseURI.

For other metadata exports, I think we removed places where CC0 was hardcoded and/or where it wasn't obvious how map custom terms/the custom license URI into the format. (I think that was noted somewhere as 'future work' but I'm not aware of anyone having taken it up.)

W.r.t. specific datasets, in cases where some files are restricted/embargoed and the terms of use/other custom terms are just about the restricted files, it may work to just put those terms on the Terms of Access for Restricted Files field - you could then have a CC0 dataset with those files having the additional Terms of Access. For cases like you have here, which only has restricted files, I'm not sure that would actually help - the metadata is already open so saying the dataset is CC0 but you can't actually get any files under CC0 terms may not make sense. (FWIW: At QDR, we opted for a 'QDR Standard license' that codifies that documentation files are CC0 and other files (that are restricted or embargoed) are subject to QDR's terms.)

**Comment by jggautier** [comment]
Thanks @qqmyers. I think it might be helpful if I look at cases where a CC0 waiver was applied and text was entered in one of the fields in the "Dataset Terms" accordion", but there are no restricted files. This is kind of just a to-do note for myself. (Of course others who read this issue and would like to review and contribute other use cases should feel free.)

One of my concerns is that if people are looking for datasets with CC0 waivers, maybe using search facets (https://github.com/IQSS/dataverse/issues/9060), or in some other way looking for datasets with no or very few use restrictions, datasets like https://doi.org/10.7910/DVN/WHNXKY won't show up. (I think the same might be true for datasets with a mix of standard licenses in a "custom license". Will people want to search for datasets whose files have a CC0 waiver, even if some of those datasets' files have other licenses? Would they be able to?)

Like you said, what's been entered in the Special Permissions field could be moved to the "Terms of Access for Restricted Files" field. Does that mean that any information about data access should be put in the "Terms of Access for Restricted Files" field or one of the other fields in the "Restricted Files + Terms of Access" accordion?:

<img width="664" alt="Screen Shot 2022-10-18 at 12 44 55 PM" src="https://user-images.githubusercontent.com/18374574/196492361-255e0be9-c6df-4221-abbf-0bb724959ce5.png">

Is the Dataverse software's model then that if a depositor wants to add information about **data use**, depositors should add that information in fields in the "Dataset Terms" accordion? And information about **data access** should go in the "Restricted Files + Terms of Access" fields?

The "Special Permissions" field is one of the fields in the "Dataset Terms" accordion and that field's description is "Determine if any special permissions are required to access a resource (e.g., if form is a needed and where to access the form)". That seems more about data access. The same seems true for the "Confidentiality Declaration" field. To maintain the model, should those fields be moved to the group of fields in the "Terms of Access for Restricted Files" accordion? Some other fields in that accordion mention both data use and access ("Restrictions" and "Conditions").

My other concern is something I've also heard from others in the community, that in 5.10+ installations, when a standard license is chosen from the list, there's no longer a place to put information that they would normally put in fields in the "Dataset Terms" accordion because those fields disappear. I think @philippconzett made a related comment in some channel recently (a GitHub issue, Slack, Google Group?), though I can't find it now. And I've also heard about complications from the Harvard repo's curation team.

I think the community should invest time in reviewing how the functionality is working, and I hope these questions could help provide some direction.

**Comment by philippconzett** [comment]
Julian, thanks for raising this issue. You're right, I have earlier commented on this, but I can't find my comment either. The comment was about the following:

When researchers have reused data from other sources and want to publish a dataset that is somehow derived from or builds on these other source, it is good practice to describe your sources and also under what Terms of Use or licenses they were (re)used. For CC BY licenses, this is even a legal requirement.

We're still running on versjon 5.6 of Dataverse, so when choosing another license than CC0, we add the Terms of Use or the standard license into the Terms of Use field in the Terms of Use tab. Into the same field, we add a description of which Terms and Use or licenses the different sources were used. See, e.g., this dataset: https://doi.org/10.18710/VMUP44. Once we implement standard license support, we won't be able to add any text to the Terms of Use field anymore. So, the question is where to put this information. The metadata field Data Sources could be an alternative, which we already have used in the Terms of Use clean-up to prepare the implementation of standard license support. But as you say, the community should invest time in agreeing on a best practice for this.

**Comment by sbarbosadataverse** [comment]
Prioritize and size*

**Comment by sbarbosadataverse** [comment]
also related: https://github.com/IQSS/dataverse/issues/8796

**Comment by cmbz** [comment]
2024/05/08
- There is a current issue that may address the [OpenAIRE export](https://dataverse.harvard.edu/api/datasets/export?exporter=oai_datacite&persistentId=doi:10.7910/DVN/WHNXKY) part of this issue
- Main task is to identify the specific programming task needed

**Comment by cmbz** [comment]
2024/06/20
- Next steps, needs additional review of work

**Comment by cmbz** [comment]
Depends on: https://github.com/IQSS/dataverse/pull/10632. It must be merged first.

**Comment by cmbz** [comment]
2025-04-07
- @jggautier now that this dependency has been merged: https://github.com/IQSS/dataverse/pull/10632 do you know what additional design/other work needs to be done to move this feature request forward?

**Comment by jggautier** [comment]
Hi @cmbz. I think additional work is more investigative and includes reviewing cases where a "standard" license is applied to a dataset but that license is not included in metadata exports, such as the metadata sent to DataCite (https://github.com/IQSS/dataverse/pull/10632) and other the metadata exports. There are two example datasets in this GitHub issue and I'm sure there are more.

Other things to consider could be the questions in [an earlier comment](https://github.com/IQSS/dataverse/issues/8798#issuecomment-1282747844) about the design of the Terms fields and [the questions that Philipp Conzett asked](https://github.com/IQSS/dataverse/issues/8798#issuecomment-1372097763).

**Comment by cmbz** [comment]
Thanks @jggautier! Who should conduct the investigation that you describe here: https://github.com/IQSS/dataverse/issues/8798#issuecomment-2786649221 ?

**Comment by jggautier** [comment]
Hmm I could when I have time. I just haven't had time.

Others who are aware of this at some level are @philippconzett and @qqmyers, who commented in this issue, and @stevenwinship and @sekmiller, who assigned themselves last year.

**Comment by jggautier** [comment]
Hmmm I'd recommend it go into a backlog.

**Comment by cmbz** [comment]
2025-04-14
- Sizing at 10 for now. The person who picks this up should resize to reflect the work needed. Also, please reach out to @jggautier and @sbarbosadataverse for information about existing use cases for this feature.

**Comment by amberleahey** [comment]
We started to do this in the updated DDI-Codebook, and PDF and HTML Exporters by updating the mapping from DV standard licenses and DV custom terms of use to DDI dataAccs > Notes > level="dv" type="DVN:TOU" see example with custom terms mapped to dataAccess: https://odesi.ca/api/ddi?id=/odesi/doi__10-5683_SP3_PO6VYU.xml and https://borealisdata.ca/dataset.xhtml?persistentId=doi:10.5683/SP3/PO6VYU > />, but agree a lot more work to be done in this area to map Dataverse Terms metadata to more standards. DDI is a good standard to review for this kind of mapping as it includes many fields that may significantly overlap with Dataverse fields.
